datafusion/core/src/physical_plan/file_format/parquet.rs

```diff
@@ -497,37 +500,67 @@ impl FileOpener for ParquetOpener {
                 &file_metrics,
             );
 
-            if enable_page_index && check_page_index_push_down_valid(&pruning_predicate) {
+            let page_index_predicates = extract_page_index_push_down_predicates(
```

Comment:
It would be nice to avoid trying to create page index predicates if `enable_page_index == false` (the feature is not enabled), but I don't think that is critical

Reply:
fix in [54c3a7d](https://github.com/apache/arrow-datafusion/pull/3967/commits/54c3a7d39d68d59b60aeb4d07bb13b0e3aedeb0a)